eax/eax_test.go

```diff
@@ -0,0 +1,367 @@
+			}
+			newData := eax.Seal(previousData, nonce, pt, header)
+			ct := newData[len(previousData):]
+			decrypted, err := eax.Open(nil, nonce, ct, header)
+			if err != nil {
+				t.Errorf(
+					`Decrypt refused valid tag (not displaying long output)`)
+					break
+			}
+			if !bytes.Equal(pt, decrypted) {
+				t.Errorf(
+					`Random Encrypt/Decrypt error (plaintexts don't match)`)
+					break
+			}
+		}
+	}
+}
+
+func TestRejectTamperedCiphertext(t *testing.T) {
+	for i := 0; i < iterations; i++ {
+		pt := make([]byte, mathrand.Intn(maxLength))
+		header := make([]byte, mathrand.Intn(maxLength))
+		key := make([]byte, blockLength)
+		nonce := make([]byte, blockLength)
+		itemsToRandomize := [][]byte{pt, header, key, nonce}
+		for _, item := range itemsToRandomize {
+			_, err := rand.Read(item)
+			if err != nil {
+				panic(err)
+			}
+		}
+		aesCipher, errAes := aes.NewCipher(key)
+		if errAes != nil {
+			panic(errAes)
+		}
+		eax, errEax := NewEAX(aesCipher)
+		if errEax != nil {
+			panic(errEax)
+		}
+		ct := eax.Seal(nil, nonce, pt, header)
+		// Change one byte of ct (could affect either the tag or the ciphertext)
+		tampered := make([]byte, len(ct))
+		copy(tampered, ct)
+		for bytes.Equal(tampered, ct) {
+			tampered[mathrand.Intn(len(ct))] = byte(mathrand.Intn(len(ct)))
+		}
+		_, err := eax.Open(nil, nonce, tampered, header)
+		if err == nil {
+			t.Errorf(`Tampered ciphertext was not refused decryption`)
+			break
+		}
+	}
+}
+
+func TestParameters(t *testing.T) {
+	t.Run("Should panic on unsupported keySize/blockSize", func(st *testing.T) {
+		keySize := mathrand.Intn(32)
+		for keySize == 16 {
+			keySize = mathrand.Intn(32)
+		}
+		key := make([]byte, keySize)
+		defer func() {
+			if r := recover(); r == nil {
+				st.Errorf("Eax didn't panic")
+			}
+		}()
+		aesCipher, errAes := aes.NewCipher(key)
+		if errAes != nil {
+			panic(errAes)
+		}
+		_, errEax := NewEAX(aesCipher)
+		// Don't panic here, let the New* procedure panic
+		if errEax != nil { }
+	})
+	t.Run("Should return error on too long tagSize", func(st *testing.T) {
+		tagSize := blockLength + 1 + mathrand.Intn(12)
+		nonceSize := 1 + mathrand.Intn(16)
+		key := make([]byte, blockLength)
+		aesCipher, errAes := aes.NewCipher(key)
+		if errAes != nil {
+			panic(errAes)
+		}
+		_, err := NewEAXWithNonceAndTagSize(aesCipher, nonceSize, tagSize)
+		if err == nil {
+			st.Errorf("No error was given")
+		}
+	})
+	t.Run("Should not give error with allowed custom parameters", func(st *testing.T) {
+		key := make([]byte, blockLength)
+		nonceSize := mathrand.Intn(32) + 1
+		tagSize := 12 + mathrand.Intn(blockLength-11)
+		aesCipher, errAes := aes.NewCipher(key)
+		if errAes != nil {
+			panic(errAes)
+		}
+		_, err := NewEAXWithNonceAndTagSize(aesCipher, nonceSize, tagSize)
+		if err != nil {
+			st.Errorf("An error was returned")
+		}
+	})
+}
+
+func BenchmarkEncrypt(b *testing.B) {
+	// OpenPGP.js defaults plaintext chunks at 256 KiB
+	plaintextLength := 262144
```

Comment:
Last nit picks:
```suggestion
	plaintextLength := 1 << 18
```
I'd also remove the reference to OpenPGP.js here since it's not particularly relevant to this implementation. You could also choose a different plaintext length if you prefer, for variety, since higher-level AEAD tests will probably already test this exact chunk length.
(Same in the next function.)

Reply:
+, setting this to `1 << 12`, a nice round number : - ) and removing JS refs, not relevant anymore.